### PR TITLE
Allow list collaborators for users with Read access to repo

### PR DESCRIPTION
**Pull request by shashvat-kedia**
Fix for issue #9956.

### CODE REVIEW

File: routers/api/v1/api.go [hunk 1]
```diff
@@ -642,11 +642,11 @@ func RegisterRoutes(m *macaron.Macaron) {
 					}, reqGitHook(), context.ReferencesGitRepo(true))
 				}, reqToken(), reqAdmin())
 				m.Group("/collaborators", func() {
-					m.Get("", repo.ListCollaborators)
-					m.Combo("/:collaborator").Get(repo.IsCollaborator).
+					m.Get("", reqAnyRepoReader(), repo.ListCollaborators)
+					m.Combo("/:collaborator", reqAdmin()).Get(repo.IsCollaborator).
 						Put(bind(api.AddCollaboratorOption{}), repo.AddCollaborator).
 						Delete(repo.DeleteCollaborator)
```
Comment (lunny) [inline routers/api/v1/api.go]: You should also add reqAdmin on `Put` and `Delete`.

**Review by lunny — CHANGES_REQUESTED** [review]

File: routers/api/v1/api.go (hunk 1 above)
Comment (davidsvantesson) [reply routers/api/v1/api.go]: Yes. I think ListCollaborators and IsCollaborator should be the same required permission.

Comment (shashvat-kedia) [reply routers/api/v1/api.go]: @davidsvantesson Yes I agree. I have made the appropriate changes.

File: routers/api/v1/api.go [hunk 2]
```diff
@@ -642,11 +642,11 @@ func RegisterRoutes(m *macaron.Macaron) {
 					}, reqGitHook(), context.ReferencesGitRepo(true))
 				}, reqToken(), reqAdmin())
 				m.Group("/collaborators", func() {
-					m.Get("", repo.ListCollaborators)
+					m.Get("", reqAnyRepoReader(), repo.ListCollaborators)
 					m.Combo("/:collaborator").Get(repo.IsCollaborator).
```
Comment (davidsvantesson) [inline routers/api/v1/api.go]:
```suggestion
					m.Group("/:collaborator"), func() {
					    m.Get(reqAnyRepoReader(), repo.IsCollaborator).
```

Comment (shashvat-kedia) [reply routers/api/v1/api.go]: @davidsvantesson I have made the change.

**Review by davidsvantesson — APPROVED** [review]

**Comment by 6543** [comment]
@sd1998 can you `make fmt` and commit the result?

File: routers/api/v1/api.go [hunk 3]
```diff
@@ -642,11 +642,13 @@ func RegisterRoutes(m *macaron.Macaron) {
 					}, reqGitHook(), context.ReferencesGitRepo(true))
 				}, reqToken(), reqAdmin())
 				m.Group("/collaborators", func() {
-					m.Get("", repo.ListCollaborators)
-					m.Combo("/:collaborator").Get(repo.IsCollaborator).
-						Put(bind(api.AddCollaboratorOption{}), repo.AddCollaborator).
-						Delete(repo.DeleteCollaborator)
-				}, reqToken(), reqAdmin())
+					m.Get("", reqAnyRepoReader(), repo.ListCollaborators)
+					m.Group("/:collaborator", func() {
```
Comment (6543) [inline routers/api/v1/api.go]: why group if you use it as combo ?!?

Comment (6543) [inline routers/api/v1/api.go]:
```suggestion
```

File: routers/api/v1/api.go [hunk 4]
```diff
@@ -642,11 +642,13 @@ func RegisterRoutes(m *macaron.Macaron) {
...
+						m.Combo("").Get(reqAnyRepoReader(), repo.IsCollaborator).
```
Comment (6543) [inline routers/api/v1/api.go]:
```suggestion
						m.Combo("/:collaborator").Get(reqAnyRepoReader(), repo.IsCollaborator).
```

File: routers/api/v1/api.go [hunk 5]
```diff
@@ -642,11 +642,13 @@ func RegisterRoutes(m *macaron.Macaron) {
...
+						Put(reqAdmin(), bind(api.AddCollaboratorOption{}), repo.AddCollaborator).
+						Delete(reqAdmin(), repo.DeleteCollaborator)
+					})
```
Comment (6543) [inline routers/api/v1/api.go]:
```suggestion
```

**Comment by davidsvantesson** [comment]
@sd1998 Do you have possibility to install gitea build tools (https://docs.gitea.io/en-us/hacking-on-gitea/) and run `make fmt`? Continuous integration is picky about style and it can be hard to get it right manually.

File: routers/api/v1/api.go (hunk 3 above)
Comment (shashvat-kedia) [reply routers/api/v1/api.go]: @6543 I have made the changes. Sorry for the delay.

**Review by lafriks — APPROVED** [review]

**Review by 6543 — APPROVED** [review]
lgtm

**Comment by 6543** [comment]
@sd1998 just update branch and CI just restart ...

**Comment by sd1998** [comment]
@6543 done.

**Comment by guillep2k** [comment]
@lunny your review is required.

**Comment by 6543** [comment]
@sd1998 can you "update" :)

**Comment by techknowlogick** [comment]
@lunny need your review

**Review by lunny — APPROVED** [review]

**Comment by techknowlogick** [comment]
ping LG-TM

**Pull request merged**